src/librbd/group/ListSnapshotsRequest.cc

```diff
@@ -0,0 +1,178 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "librbd/group/ListSnapshotsRequest.h"
+#include "include/ceph_assert.h"
+#include "common/dout.h"
+#include "common/errno.h"
+#include "common/ceph_context.h"
+#include "cls/rbd/cls_rbd_client.h"
+#include "librbd/Utils.h"
+
+#define dout_subsys ceph_subsys_rbd
+#undef dout_prefix
+#define dout_prefix *_dout << "librbd::group::ListSnapshotsRequest: " << this \
+                           << " " << __func__ << ": "
+
+namespace librbd {
+namespace group {
+
+using util::create_rados_callback;
+
+namespace {
+
+static const uint32_t MAX_RETURN = 1024;
+
+} // anonymous namespace
+
+template <typename I>
+void ListSnapshotsRequest<I>::send() {
+  list_snap_orders();
+}
+
+template <typename I>
+void ListSnapshotsRequest<I>::list_snap_orders() {
+  if (!m_sort) {
+    list_snapshots();
+    return;
+  }
+
+  CephContext *cct = (CephContext *)m_group_io_ctx.cct();
+  ldout(cct, 10) << dendl;
+
+  librados::ObjectReadOperation op;
+  cls_client::group_snap_list_order_start(&op, m_start_after_order, MAX_RETURN);
+  auto comp = create_rados_callback< ListSnapshotsRequest<I>,
+                   &ListSnapshotsRequest<I>::handle_list_snap_orders>(this);
```

Comment:
I don't understand what is mean by spell out here - please clarify.

Reply:
`create_rados_callback` -> `util::create_rados_callback`

Reply:
Why isn't this aligned with util?

Reply:
It's formatted as a list of arguments in a function call would be.  Generally, the rule for function calls is: if aligning with the first argument isn't possible because one of the arguments is too long
```
auto result = DoFoo(arg1, arg2,
                    verylongarg3,
                    arg4);
```
then all arguments are placed on subsequent lines with double indent (i.e. 4 spaces)
```
auto result = DoFoo(
    arg1, arg2, verylongarg3, arg4);
```
```
auto result = DoFoo(
    arg1, arg2,
    verylongarg3, arg4);
```
Now, librbd mostly ignores the "double indent" part of the rule, but the "all arguments on subsequent lines" part is followed.  For example:
```
    auto req = ObjectDispatchSpec::create_read(
      &image_ctx, OBJECT_DISPATCH_LAYER_NONE, oe.object_no,
      &req_comp->extents, m_io_context, m_op_flags, m_read_flags,
      this->m_trace, nullptr, req_comp);
```